src/surveys/components/RangeResponse.tsx

```diff
@@ -0,0 +1,145 @@
+import classnames from 'classnames';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import { Button, Header } from '../../components';
+import { ID, Question, RangeSettings, SurveyAnswer, Theme } from '../../types';
+import RatingIcon from './RatingIcon';
+
+interface RangeResponseProps {
+  question: Question;
+  onAnswered: (questionId: ID, answer: SurveyAnswer) => void;
+  theme?: Theme;
+}
+
+function RangeResponse({ question, onAnswered, theme }: RangeResponseProps) {
+  const [value, setValue] = useState(0);
+  const [answerId, setAnswerId] = useState<ID | null>(null);
+
+  const lightAnswerColor = theme?.answer && `${theme?.answer}1A`;
+  const mediumAnswerColor = theme?.answer && `${theme?.answer}9A`;
```

Comment:
@AfeezGL I'm not really familiar with Tailwind but I'm guessing `9A` means 100%, right?

Reply:
@jdamilola No, that's 60% and the `1A` is 10% opacity. For the NPS buttons, the background color is `theme.answer` with 10% opacity and the selected option has 60% opacity. That was done to make the button text a little visible since the button text shares the same color with button background.
![Screenshot (16)](https://github.com/Formily/formily-web/assets/45849676/15b61d3a-45e9-4929-b4d8-ce878231bcec)

Reply:
@AfeezGL The text color should be 100% while the background color should be 10%. The hover and selected state should be 20%. See below screenshot.

<img width="525" alt="Screenshot 2023-08-26 at 00 58 40" src="https://github.com/Formily/formily-web/assets/113894296/967d0ecf-c54d-43fc-9d46-6d0cee5fdfac">

Reply:
@AfeezGL I also feel the spacing between button is too wide. Is this what we have on Figma?

Reply:
@jdamilola It was `18px` on figma but I'm using `16px`, I think it was looking off because I zoomed in on the screenshot.